fronts-client/src/util/collectionUtils.ts

```diff
@@ -37,27 +37,29 @@ export type MappableDropType =
   | ChefDrop
   | FeastCollectionDrop;
 
-const dropToCard = (e: React.DragEvent): MappableDropType | null => {
-  const map = {
-    capi: (data: string): CAPIDrop => ({
-      type: 'CAPI',
-      data: JSON.parse(data),
-    }),
-    [CardTypesMap.RECIPE]: (data: string): RecipeDrop => ({
-      type: 'RECIPE',
-      data: JSON.parse(data),
-    }),
-    [CardTypesMap.CHEF]: (data: string): ChefDrop => ({
-      type: 'CHEF',
-      data: JSON.parse(data),
-    }),
-    [CardTypesMap.FEAST_COLLECTION]: (data: string): FeastCollectionDrop => ({
-      type: 'FEAST_COLLECTION',
-    }),
-    text: (url: string): RefDrop => ({ type: 'REF', data: url }),
-  };
+const dropToCardMap = {
```

Comment:
Moved to its own data structure so we could statically derive its keys, and get some type safety upstream.